piqueserver/config/scripts/ratio.py

```diff
@@ -5,47 +5,75 @@
 Maintainer: mat^2
 """
 
-from piqueserver.commands import get_player, add
+from commands import get_player, add
+from pyspades.constants import *
+
+# True if you want to include the headshot-death ratio in the ratio
+# NOTE: this makes the message overflow into two lines
+HEADSHOT_RATIO = False
+
+# List other types of kills as well
+EXTENDED_RATIO = False
 
 # "ratio" must be AFTER "votekick" in the config.txt script list
 RATIO_ON_VOTEKICK = True
 IRC_ONLY = False
 
 
 def ratio(connection, user=None):
-    has_msg = "You have"
+    msg = "You have"
     if user != None:
         connection = get_player(connection.protocol, user)
-        has_msg = "%s has"
+        msg = "%s has"
         if connection not in connection.protocol.players:
             raise KeyError()
-        has_msg %= connection.name
+        msg %= connection.name
     if connection not in connection.protocol.players:
         raise KeyError()
-    ratio = connection.ratio_kills / float(max(1, connection.ratio_deaths))
-    ratio_msg = has_msg + (" a kill-death ratio of %.2f" % (ratio))
-    return ('%s (%s kills, %s deaths).' %
-            (ratio_msg, connection.ratio_kills, connection.ratio_deaths))
+
+    kills = connection.ratio_kills
+    deaths = float(max(1, connection.ratio_deaths))
+    headshotkills = connection.ratio_headshotkills
+    meleekills = connection.ratio_meleekills
+    grenadekills = connection.ratio_grenadekills
+
+    msg += " a kill-death ratio of %.2f" % (kills / deaths)
+    if HEADSHOT_RATIO:
+        msg += ", headshot-death ratio of %.2f" % (headshotkills / deaths)
+    msg += " (%s kills, %s deaths" % (kills, connection.ratio_deaths)
+    if EXTENDED_RATIO:
+        msg += ", %s headshot, %s melee, %s grenade" % (
+            headshotkills, meleekills, grenadekills)
+    msg += ")."
+    return msg
 
 add(ratio)
 
 
 def apply_script(protocol, connection, config):
     class RatioConnection(connection):
         ratio_kills = 0
+        ratio_headshotkills = 0
+        ratio_meleekills = 0
+        ratio_grenadekills = 0
         ratio_deaths = 0
 
         def on_kill(self, killer, type, grenade):
             if killer is not None and self.team is not killer.team:
                 if self != killer:
                     killer.ratio_kills += 1
+                    killer.ratio_headshotkills += type == HEADSHOT_KILL
```

Comment:
It's adding the result of `type == HEADSHOT_KILL` to `killer.ratio_headshotkills`. As `True == 1`, this will increment `ratio_headshotkills` by one if it's a headshot.